frontend/packages/console-shared/src/hooks/useUserSettingsLocalStorage.ts

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+// FIXME upgrading redux types is causing many errors at this time
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useSelector } from 'react-redux';
+import { RootState } from '@console/internal/redux';
+import { deseralizeData, seralizeData } from '../utils/user-settings';
+
+const CONFIGMAP_LS_KEY = 'console-user-settings';
+
+export const useUserSettingsLocalStorage = <T>(
+  key: string,
+  defaultValue: T,
+  watch: boolean = true,
+): [T, React.Dispatch<React.SetStateAction<T>>] => {
+  const keyRef = React.useRef(key);
+  const defaultValueRef = React.useRef(defaultValue);
+  const userUid = useSelector(
+    (state: RootState) => state.UI.get('user')?.metadata?.uid ?? 'kubeadmin',
+  );
+  const storageConfigNameRef = React.useRef(`${CONFIGMAP_LS_KEY}-${userUid}`);
+  const [lsData, setLsData] = React.useState(() => {
+    const valueInLocalStorage = deseralizeData(localStorage.getItem(storageConfigNameRef.current));
```

Comment:
You could use `localStorage.hasItem` to check whether we fallback to the default or not

Reply:
`hasItem` i din't see in API for localStorage so made a check against null

```
localStorage.getItem(storageConfigNameRef.current) !== null
```